File: bitwise_problems/xor_from_0_to_n/solution.js
// time complexity---O(1)
const xor=(num)=>
    {
        if(num%4==0)
            {
                return num;
            }
        else if(num%4==1)
            {
                return 1;
            }
        else if(num%4==2)
            {
                return num+1;
            }
        else{
            return 0
        }

    };

console.log(xor(3));

File: bitwise_problems/xor_from_0_to_n/solution.py
'''
time complexity--O(1)
'''
def xor(num):
    if num%4==0:
        return num
    elif num%4==1:
        return 1
    elif num%4==2:
        return num+1
    elif num%3==0:
        return 0
    
print(xor(9))